tensorflow/contrib/tensorrt/convert/convert_nodes.cc

```diff
@@ -1577,6 +1712,104 @@ tensorflow::Status ConvertActivation(
   return tensorflow::Status::OK();
 }
 
+tensorflow::Status ConvertQuantize(
+    Converter& ctx, const tensorflow::NodeDef& node_def,
+    const std::vector<TRT_TensorOrWeights>& inputs,
+    std::vector<TRT_TensorOrWeights>* outputs) {
+  if (inputs.at(0).is_weights()) {
+    outputs->push_back(inputs.at(0));
+    return tensorflow::Status::OK();
+  }
+  // ***************************************************************************
+  // Fake quantization nodes are replaced with no-ops and their min/max values
+  // are applied as the quantization scale for the input and output
+  // ***************************************************************************
+  nvinfer1::ITensor* tensor = const_cast<nvinfer1::ITensor*>(inputs.at(0).tensor());
+  // Min
+  TRT_ShapedWeights weights_min = inputs.at(1).weights();
+  auto weights_min_ptr = static_cast<float*>(const_cast<void*>(weights_min.GetValues()));
+  float min_range = weights_min_ptr[0];
+  // Max
+  TRT_ShapedWeights weights_max = inputs.at(2).weights();
+  auto weights_max_ptr = static_cast<float*>(const_cast<void*>(weights_max.GetValues()));
+  float max_range = weights_max_ptr[0];
+  
+  // The default shuffle layer serves as a dummy layer which will be optimized
+  // away. Identity layer does not get optimized away as of TRT 5.0, but
+  // once we know that it does it would be nice to use that instead.
+  nvinfer1::IShuffleLayer* layer;
+  layer = ctx.network()->addShuffle(*tensor);
+  TFTRT_RETURN_ERROR_IF_NULLPTR(layer, node_def.name());
+  nvinfer1::ITensor* output_tensor = layer->getOutput(0);
+
+  // Store ranges for input and output of dummy op
+  ctx.ProvideQuantizationRange(tensor, min_range, max_range);
```

Comment:
Here we assume that `tensor` is not used elsewhere as input to other nodes. If user has a crazy graph where `tensor` is fed into different ops and some but not all of them are QDQV2, or the ranges of the QDQV2s are different, there will be problem. I don't see any easy way to detect this with the current implementation, so may be add comment and a `LOG(WARNING)`?

Do you have a better idea? Also @wujingyue

Reply:
You would need mixed-precision to solve these cases. One potential solution is to keep the tensor in fp and quantize it on one path bot not the other, or quantize it differently on both paths. 

Given TensorRT doesn't plan to support mixed-precision in short term, I'd emit a fatal error if this happens. We can make sure quantized training doesn't produce graphs like that at the first place.